File: xdef.net/xdef.net/Connection/RemoteHandlingObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace xdef.net.Connection
{
    internal abstract class RemoteHandlingObject : RemoteObject
    {
        public RemoteHandlingObject(Client client) : base(0, client)
        {
        }

        protected override void DeleteRemoteObject()
        {
            // Do nothing
        }

        public abstract Request HandleRequest(Request request);
    }
}

File: xdef.bridge/xdef.bridge/src/main/java/org/xdef/bridge/remoteObjects/RemoteHandlingObject.java
package org.xdef.bridge.remoteObjects;

import org.xdef.bridge.server.Client;
import org.xdef.bridge.server.requests.Request;
import org.xdef.bridge.server.requests.Response;

public abstract class RemoteHandlingObject extends RemoteObject {

    public RemoteHandlingObject(Client client) {
        super(client, 0);
    }

    public abstract Response handleRequest(Request request);
    
    @Override
    protected void deleteRemoteObject() {
        // Do nothing
    }
}